iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/ReportedPropertiesTests.java

```diff
@@ -153,31 +157,31 @@ public void testUpdateReportedPropertiesMultiThreaded() throws IOException, Inte
         // send max_prop RP all at once
         deviceUnderTest.dCDeviceForTwin.createNewReportedProperties(MAX_PROPERTIES_TO_TEST);
         internalClient.sendReportedProperties(deviceUnderTest.dCDeviceForTwin.getReportedProp());
-        Thread.sleep(DELAY_BETWEEN_OPERATIONS);
+        Thread.sleep(REPORTED_PROPERTIES_PROPAGATION_DELAY_MILLISECONDS);
 
         // act
         // Update RP
         for (int i = 0; i < MAX_PROPERTIES_TO_TEST; i++)
         {
-        final int index = i;
-        executor.submit(new Runnable()
-        {
-            @Override
-            public void run()
+            final int index = i;
+            executor.submit(new Runnable()
             {
-                try
-                {
-                    deviceUnderTest.dCDeviceForTwin.updateExistingReportedProperty(index);
-                    internalClient.sendReportedProperties(deviceUnderTest.dCDeviceForTwin.getReportedProp());
-                    waitAndVerifyTwinStatusBecomesSuccess();
-                }
-                catch (IOException | InterruptedException e)
+                @Override
+                public void run()
                 {
-                    fail(CorrelationDetailsLoggingAssert.buildExceptionMessage("Unexpected exception occurred during sending reported properties: " + Tools.getStackTraceFromThrowable(e), internalClient));
+                    try
+                    {
+                        Set<Property> updatedProperties = deviceUnderTest.dCDeviceForTwin.updateExistingReportedProperty(index);
+                        internalClient.sendReportedProperties(updatedProperties);
```

Comment:
Again, only send the reported properties that belong to this thread. See above for why